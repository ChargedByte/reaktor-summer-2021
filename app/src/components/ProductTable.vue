<template>
  <v-data-table
    :footer-props="footerProps"
    :headers="headers"
    :items="items"
    disable-filtering
    disable-sort
  >
    <template #item.colors="{ item }">
      <ColorsItem :value="item.colors" />
    </template>

    <template #item.availability="{ item }">
      <AvailabilityChip :value="item.availability" />
    </template>
  </v-data-table>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator'

import AvailabilityChip from '~/components/AvailabilityChip.vue'
import ColorsItem from '~/components/ColorsItem.vue'

import Product from '~/model/product'

@Component({
  components: { ColorsItem, AvailabilityChip },
})
export default class ProductTable extends Vue {
  @Prop()
  items!: Array<Product>

  footerProps = {
    itemsPerPageOptions: [5, 10, 15, 25, 50, -1],
  }

  headers = [
    {
      text: 'Id',
      value: 'id',
      width: '20%',
      divider: true,
    },
    {
      text: 'Name',
      value: 'name',
      width: '25%',
      divider: true,
    },
    {
      text: 'Manufacturer',
      value: 'manufacturer',
      width: '15%',
      divider: true,
    },
    {
      text: 'Price',
      value: 'price',
      width: '10%',
      divider: true,
    },
    {
      text: 'Color(s)',
      value: 'colors',
      width: '20%',
      divider: true,
    },
    {
      text: 'Availability',
      value: 'availability',
      width: '10%',
    },
  ]
}
</script>
