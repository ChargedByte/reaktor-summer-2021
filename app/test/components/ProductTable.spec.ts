import { createLocalVue, mount } from '@vue/test-utils'

import Vuetify from 'vuetify'

import ProductTable from '~/components/ProductTable.vue'

import { beanies } from '~~/test/__mocks__/products'

describe('ProductTable', () => {
  const localVue = createLocalVue()
  let vuetify: Vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  test('is a Vue instance', () => {
    const wrapper = mount(ProductTable, {
      localVue,
      vuetify,
      propsData: { items: [] },
    })
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    const wrapper = mount(ProductTable, {
      localVue,
      vuetify,
      propsData: { items: beanies },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
