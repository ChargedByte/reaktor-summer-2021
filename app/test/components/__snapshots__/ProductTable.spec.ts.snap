// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTable renders correctly 1`] = `
"<div class=\\"v-data-table v-data-table--has-bottom theme--light\\">
  <div class=\\"v-data-table__wrapper\\">
    <table>
      <colgroup>
        <col class=\\"divider\\">
        <col class=\\"divider\\">
        <col class=\\"divider\\">
        <col class=\\"divider\\">
        <col class=\\"divider\\">
        <col class=\\"\\">
      </colgroup>
      <thead class=\\"v-data-table-header\\">
        <tr>
          <th role=\\"columnheader\\" scope=\\"col\\" aria-label=\\"Id\\" class=\\"text-start v-data-table__divider\\" style=\\"width: 20%; min-width: 20%;\\"><span>Id</span></th>
          <th role=\\"columnheader\\" scope=\\"col\\" aria-label=\\"Name\\" class=\\"text-start v-data-table__divider\\" style=\\"width: 25%; min-width: 25%;\\"><span>Name</span></th>
          <th role=\\"columnheader\\" scope=\\"col\\" aria-label=\\"Manufacturer\\" class=\\"text-start v-data-table__divider\\" style=\\"width: 15%; min-width: 15%;\\"><span>Manufacturer</span></th>
          <th role=\\"columnheader\\" scope=\\"col\\" aria-label=\\"Price\\" class=\\"text-start v-data-table__divider\\" style=\\"width: 10%; min-width: 10%;\\"><span>Price</span></th>
          <th role=\\"columnheader\\" scope=\\"col\\" aria-label=\\"Color(s)\\" class=\\"text-start v-data-table__divider\\" style=\\"width: 20%; min-width: 20%;\\"><span>Color(s)</span></th>
          <th role=\\"columnheader\\" scope=\\"col\\" aria-label=\\"Availability\\" class=\\"text-start\\" style=\\"width: 10%; min-width: 10%;\\"><span>Availability</span></th>
        </tr>
      </thead>
      <tbody>
        <tr class=\\"\\">
          <td class=\\"text-start v-data-table__divider\\">R4OuKi9VnVwMKy321ikA</td>
          <td class=\\"text-start v-data-table__divider\\">habitasse platea</td>
          <td class=\\"text-start v-data-table__divider\\">nascetur</td>
          <td class=\\"text-start v-data-table__divider\\">638</td>
          <td class=\\"text-start v-data-table__divider\\"><span>Khaki, Fuscia</span></td>
          <td class=\\"text-start\\"><span class=\\"v-chip theme--light v-size--small success\\"><span class=\\"v-chip__content\\">In Stock</span></span></td>
        </tr>
        <tr class=\\"\\">
          <td class=\\"text-start v-data-table__divider\\">6rCPg5CcqFJcpAWZDXU0</td>
          <td class=\\"text-start v-data-table__divider\\">mattis egestas</td>
          <td class=\\"text-start v-data-table__divider\\">aenean</td>
          <td class=\\"text-start v-data-table__divider\\">614</td>
          <td class=\\"text-start v-data-table__divider\\"><span>Pink, Fuscia, Crimson</span></td>
          <td class=\\"text-start\\"><span class=\\"v-chip theme--light v-size--small success\\"><span class=\\"v-chip__content\\">In Stock</span></span></td>
        </tr>
        <tr class=\\"\\">
          <td class=\\"text-start v-data-table__divider\\">GgFnGfzmkZatPp6Sm17p</td>
          <td class=\\"text-start v-data-table__divider\\">mauris eget massa</td>
          <td class=\\"text-start v-data-table__divider\\">tincidunt</td>
          <td class=\\"text-start v-data-table__divider\\">305</td>
          <td class=\\"text-start v-data-table__divider\\"><span>Purple</span></td>
          <td class=\\"text-start\\"><span class=\\"v-chip theme--light v-size--small error\\"><span class=\\"v-chip__content\\">Out of Stock</span></span></td>
        </tr>
        <tr class=\\"\\">
          <td class=\\"text-start v-data-table__divider\\">chUDARuvQnB021wTfEeO</td>
          <td class=\\"text-start v-data-table__divider\\">libero quis</td>
          <td class=\\"text-start v-data-table__divider\\">tortor</td>
          <td class=\\"text-start v-data-table__divider\\">528</td>
          <td class=\\"text-start v-data-table__divider\\"><span>Puce, Puce</span></td>
          <td class=\\"text-start\\"><span class=\\"v-chip theme--light v-size--small warning\\"><span class=\\"v-chip__content\\">Less than 10</span></span></td>
        </tr>
        <tr class=\\"\\">
          <td class=\\"text-start v-data-table__divider\\">DenjHhqoM2UXePNNY0y1</td>
          <td class=\\"text-start v-data-table__divider\\">turpis elementum ligula</td>
          <td class=\\"text-start v-data-table__divider\\">justo</td>
          <td class=\\"text-start v-data-table__divider\\">426</td>
          <td class=\\"text-start v-data-table__divider\\"><span>Khaki</span></td>
          <td class=\\"text-start\\"><span class=\\"v-chip theme--light v-size--small success\\"><span class=\\"v-chip__content\\">In Stock</span></span></td>
        </tr>
        <tr class=\\"\\">
          <td class=\\"text-start v-data-table__divider\\">1XexnO7fo9opvgtCWAFE</td>
          <td class=\\"text-start v-data-table__divider\\">nam dui</td>
          <td class=\\"text-start v-data-table__divider\\">ullamcorper</td>
          <td class=\\"text-start v-data-table__divider\\">161</td>
          <td class=\\"text-start v-data-table__divider\\"><span>Teal</span></td>
          <td class=\\"text-start\\"><span class=\\"v-chip theme--light v-size--small error\\"><span class=\\"v-chip__content\\">Out of Stock</span></span></td>
        </tr>
        <tr class=\\"\\">
          <td class=\\"text-start v-data-table__divider\\">TzpkfyvDkthZmmgq62S6</td>
          <td class=\\"text-start v-data-table__divider\\">diam erat</td>
          <td class=\\"text-start v-data-table__divider\\">interdum</td>
          <td class=\\"text-start v-data-table__divider\\">585</td>
          <td class=\\"text-start v-data-table__divider\\"><span>Puce, Teal</span></td>
          <td class=\\"text-start\\"><span class=\\"v-chip theme--light v-size--small success\\"><span class=\\"v-chip__content\\">In Stock</span></span></td>
        </tr>
        <tr class=\\"\\">
          <td class=\\"text-start v-data-table__divider\\">iMmLiRxw6tQ3eGeCO9I0</td>
          <td class=\\"text-start v-data-table__divider\\">pede posuere</td>
          <td class=\\"text-start v-data-table__divider\\">orci</td>
          <td class=\\"text-start v-data-table__divider\\">540</td>
          <td class=\\"text-start v-data-table__divider\\"><span>Mauv, Yellow</span></td>
          <td class=\\"text-start\\"><span class=\\"v-chip theme--light v-size--small error\\"><span class=\\"v-chip__content\\">Out of Stock</span></span></td>
        </tr>
        <tr class=\\"\\">
          <td class=\\"text-start v-data-table__divider\\">R00IGqYRYA6OEJyzmQOh</td>
          <td class=\\"text-start v-data-table__divider\\">volutpat in congue</td>
          <td class=\\"text-start v-data-table__divider\\">nonummy</td>
          <td class=\\"text-start v-data-table__divider\\">148</td>
          <td class=\\"text-start v-data-table__divider\\"><span>Red, Yellow, Violet</span></td>
          <td class=\\"text-start\\"><span class=\\"v-chip theme--light v-size--small success\\"><span class=\\"v-chip__content\\">In Stock</span></span></td>
        </tr>
        <tr class=\\"\\">
          <td class=\\"text-start v-data-table__divider\\">r5fD2JsKobkwAOQ7yMLX</td>
          <td class=\\"text-start v-data-table__divider\\">ut erat</td>
          <td class=\\"text-start v-data-table__divider\\">condimentum</td>
          <td class=\\"text-start v-data-table__divider\\">122</td>
          <td class=\\"text-start v-data-table__divider\\"><span>Orange, Violet, Indigo</span></td>
          <td class=\\"text-start\\"><span class=\\"v-chip theme--light v-size--small error\\"><span class=\\"v-chip__content\\">Out of Stock</span></span></td>
        </tr>
      </tbody>
    </table>
  </div>
  <div class=\\"v-data-footer\\">
    <div class=\\"v-data-footer__select\\">Rows per page:<div class=\\"v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select\\">
        <div class=\\"v-input__control\\">
          <div role=\\"button\\" aria-haspopup=\\"listbox\\" aria-expanded=\\"false\\" aria-owns=\\"list-51\\" class=\\"v-input__slot\\">
            <div class=\\"v-select__slot\\">
              <div class=\\"v-select__selections\\">
                <div class=\\"v-select__selection v-select__selection--comma\\">10</div><input aria-label=\\"Rows per page:\\" id=\\"input-51\\" readonly=\\"readonly\\" type=\\"text\\" aria-readonly=\\"false\\" autocomplete=\\"off\\">
              </div>
              <div class=\\"v-input__append-inner\\">
                <div class=\\"v-input__icon v-input__icon--append\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-menu-down theme--light\\"></i></div>
              </div><input type=\\"hidden\\" value=\\"10\\">
            </div>
            <div class=\\"v-menu\\">
              <!---->
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class=\\"v-data-footer__pagination\\">1-10 of 10</div>
    <div class=\\"v-data-footer__icons-before\\"><button type=\\"button\\" disabled=\\"disabled\\" class=\\"v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default\\" aria-label=\\"Previous page\\"><span class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chevron-left theme--light\\"></i></span></button></div>
    <div class=\\"v-data-footer__icons-after\\"><button type=\\"button\\" disabled=\\"disabled\\" class=\\"v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default\\" aria-label=\\"Next page\\"><span class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chevron-right theme--light\\"></i></span></button></div>
  </div>
</div>"
`;
